### PR TITLE
Add Terraform Config for provisioning aks cluster 

### PR DESCRIPTION
**Pull request by sheneska**
### Description

This PR created a new terraform configuration for provisioning aks clusters.

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: kubernetes/test-infra/aks-new/main.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
+provider "azurerm" {
+  features {}
+}
+
+variable "location" {
+  type = string
+  default = "West Europe"
```
Comment (BBBmau) [inline kubernetes/test-infra/aks-new/main.tf:7]: Attempted to run with `West Europe`, we currently have reached our quota for this region. Changing it to `West US` works just fine.

File: kubernetes/test-infra/aks-new/main.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
+provider "azurerm" {
+  features {}
```
Comment (BBBmau) [inline kubernetes/test-infra/aks-new/main.tf:2]: when attempting to use `export ARM_SUBSCRIPTION_ID` it doesn't seem to apply the id correctly. Using the attribute `subscription_id` within the provider block solves this issue.

**Review by BBBmau — CHANGES_REQUESTED** [review]
After doing the requested changes, i was able to provision the cluster successfully!

**Comment by BBBmau** [comment]
be sure to add the copyright headers at the top of the file as well!

```
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0
```

**Review by jrhouston — APPROVED** [review]

**Pull request merged**